analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Invocation.cs

```diff
@@ -1187,4 +1187,29 @@ public void Main()
         validator.TagValue("Args2").Should().HaveOnlyConstraint(ObjectConstraint.NotNull);
         validator.TagValue("Args3").Should().HaveOnlyConstraint(ObjectConstraint.NotNull);
     }
+
+    [DataTestMethod]
+    [DataRow("null", 0)]
+    [DataRow("42", 42)]
+    public void Invocation_NullableGetHasValue_LearnsBoolConstraint(string value, int expected)
+    {
+        var code = $"""
+            int? value = {value};
+            value ??= 0;    // Uses InvocationOperation int?.HasValue.get()
+            Tag("Value", value);
+            """;
+        SETestContext.CreateCS(code).Validator.TagValue("Value").Should().HaveOnlyConstraints(ObjectConstraint.NotNull, NumberConstraint.From(expected));
+    }
+
+    [TestMethod]
+    public void Invocation_NullableGetHasValue_Unknown()
+    {
+        const string code = """
+            arg ??= 0;      // Uses InvocationOperation int?.HasValue.get()
+            Tag("Arg", arg);
+            """;
+        SETestContext.CreateCS(code, "int? arg").Validator.TagValues("Arg").Should().SatisfyRespectively(
+            x => x.Should().HaveNoConstraints(),
```

Comment:
What would be necessary to learn `NotNull` for this branch as well? Is this planned?

Reply:
It's practically easy to do. 

And pointless at the same time. As this weird `int?.HasValue.get()` cannot be directly invoked by user. It's just internal representation on the CFG level. And I've seen it only in `??=`.

Thinking out loud: 
Does it make sense to split the flow to `arg Null && False` and `arg NotNull && True`?
`arg Null && False` goes into `??=` assignment. Likely getting `arg NotNull`
`arg NotNull && True` follows with `arg NotNull`

Without this, `arg` has no constraints. While we've null-tested.

Reply:
So it actually makes sense to add it because of cases like
```
arg ??= 0;
if(arg is null)
{
  // This should be unreachable
}
```

Reply:
Yes, exactly. I created an issue:
https://github.com/SonarSource/sonar-dotnet/issues/7322